### DISCUSSION TITLE
Rotary Push Encoder & OpenDeck

### DISCUSSION THREAD

**Discussion by poncratias** [shanteacontrols/OpenDeck#122] [Q&A]
Hi,
i am totally new to arduino, midi controllers etc, but thanks to OpenDeck i was able to build my own midicontroller with knobs & potentiometers! nice!

now i just have one problem: i want to use a rotary push encoder.

It got the pins labeld GND, + , SW, DT, CLK

now how or where should i connect these on my opendeck board? i just dont get it.... 

thx for any help!

**Comment by paradajz** [comment]
Unfortunately such encoders aren't supported.

**Reply by toyoto** [reply]
Hello Paradajz, is it now possible to use rotary encoder with push? Is there a way to wire the push on a digital input?

**Reply by paradajz** [reply]
Yes it is. Just follow the pinout - those encoders have 2 encoder pins (A+B), ground, and 2 extra pins for button (in most cases). Even the encoder from the original question had this wiring scheme so my answer is incorrect - you just need to find the pinout of the encoder in question.

**Reply by toyoto** [reply]
This is great! Thanks

**Reply by FractalFM3** [reply]
Hello Paradajz, 
in the Opendeck M wiki doc the pinout for encoder is Pin1 Pin2 and PIN3.
Should A go to Pin1 and B to pin 3 with ground to Pin2 ? I have hard time to find example of connection for the encoders to the Pins listed in the wiki, for button and Potentiometer it is ok but Encoder, I am not sure.
Thanks in advance

**Reply by paradajz** [reply]
The middle pin on encoder goes to colum, and two outer pins go to rows.

> Should A go to Pin1 and B to pin 3 with ground to Pin2 ?

Pin 3 doesn't go to ground but to column pin instead, as mentioned in the doc.
